### ISSUE TITLE
Investigate and address any accessibility violations in stable components

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#7527]
### Background
A handful of our stable components currently show violations under the Accessibility tab.

![Image](https://github.com/user-attachments/assets/e4714117-f2b6-4e81-8c0a-cce33e4994ce)

### The solution
Document and resolve any current violations on stable components. This work does not include fixes for “preview” / “preview candidate” components or patterns.

### Acceptance criteria
- [x] Identify components with active violations in the accessibility tab
  - e.g. Productive card clickable variant (can remove this by removing `clickZone: one` or by removing the interactive ”Read more” from the Storybook example.
- [x] Document and organize “incomplete” violations in stable components in a new followup issue

**Comment by makafsal** [comment]
# Investigation report

| Component | Fixed violations | Pending violations | Note |
| ------------ | --------------- | ----------------- | ----- |
| `AboutModal` | **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_ <hr /> **2. Element with "region" role does not have a label**<br /><br />💡 _Opened an [issue](https://github.com/carbon-design-system/carbon/issues/19512) in the core_<br /><br /> | ❌ 1 | Fixed `1` existing issues<br />Remaining `1` violations will be addressed in this  [issue](https://github.com/carbon-design-system/carbon/issues/19512) |
| `ExpressiveCard` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<br /><br /> | ✅ None | Fixed `1` existing issue |
| `ProductiveCard` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<br /><br /> | ✅ None | Fixed `1` existing issue |
| `ExportModal` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<hr />**2.Group/Fieldset does not have an accessible name**<br /><br />💡 _Resolved by providing `aria-label='extensions'` for `RadioButtonGroup` No need for a prop to provide `aria-label` because it should always be `extensions`_<br /><br /> | ✅ None | Fixed `2` existing issues |
| `FullPageError` |  **No existing violations** | ✅ None | - |
| `APIKeyModal` |  **1. Content is not within a landmark element** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<hr />**2. Group/Fieldset does not have an accessible name**<br /><br />💡 _Resolved by providing `aria-label` for `RadioButtonGroup` in the story_<hr />**3. Element with "region" role does not have a label**<br /><br />💡 _Opened an [issue](https://github.com/carbon-design-system/carbon/issues/19512) in the core_ <hr /> **4. The tabbable element's role 'region' is not a widget role**<br /><br />💡 _Opened an [issue](https://github.com/carbon-design-system/carbon/issues/19512) in the core_<br /><br /> | ❌ 2 | Fixed `2` existing issue.<br /><br />Remaining `2` violations will be addressed in this  [issue](https://github.com/carbon-design-system/carbon/issues/19512) |
| `NotificationPanel` |  **1. Content is not within a landmark element** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<br /><br />**2. Ensure interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies** <br /><br />💡 _Raised a separate [issue](https://github.com/carbon-design-system/ibm-products/issues/7612) for this because it needs a rework in the `NotificationPanel` items DOM structure_<br /><br /> | ❌ 1 | Fixed `1` issue and opened an [issue](https://github.com/carbon-design-system/ibm-products/issues/7612) for remaining `1` |
| `OptionsTile` |  **No existing violations** | ✅ None | - |
| `PageHeader` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the `PageHeader` component in a `main` tag in stories_<hr />**2. The ARIA attributes "aria-label" are not valid for the element <div> with implicit ARIA role "generic"**<br/><br />💡 _Raised an [issue](https://github.com/carbon-design-system/carbon/issues/19517) in the core repo_<br /><br /> | ❌ 1 | Fixed `1` existing issue.<br />Remaining `1` will be addressed in this [issue](https://github.com/carbon-design-system/carbon/issues/19517) |
| `Saving` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<br /><br /> | ✅ None | Fixed `1` existing issue |
| `SidePanel` |  **1. Element with "complementary" role does not have a label** (Story: Without title) <br /><br />💡 _Resolved by providing `aria-label='SidePanel without title'` in the Without title story_<br /><br /> | ✅ None | Fixed `1` existing issue |
| `TagSet` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)**<br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<hr />**2. The tabbable element's role 'region' is not a widget role** (When opening the view all modal)<br /><br />💡 _Opened an [issue](https://github.com/carbon-design-system/carbon/issues/19512) in the core_<br /><br /> | ❌ 1 | Fixed `1` existing issue<br />Remaining `1` violations will be addressed in this  [issue](https://github.com/carbon-design-system/carbon/issues/19512) |
| `Tearsheet` |  **1. Content is not within a landmark element** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<br /><br /> | ✅ None | Fixed `1` existing issue |
| `UserAvatar` |  **1. The page does not provide a way to quickly navigate to the main content (ARIA "main" landmark or a skip link)** <br /><br />💡 _Resolved by wrapping the content with `main` tag in stories_<br /><br /> | ✅ None | Fixed `1` existing issue |
